src/NLog/LayoutRenderers/ExceptionLayoutRenderer.cs

```diff
@@ -268,52 +293,54 @@ protected virtual void AppendData(StringBuilder sb, Exception ex)
             }
         }
 
-        private ExceptionDataTarget[] CompileFormat(string formatSpecifier)
+        /// <summary>
+        /// Split the string and then compile into list of Rendering formats.
+        /// </summary>
+        /// <param name="formatSpecifier"></param>
+        /// <returns></returns>
+        private List<ExceptionRenderingFormat> CompileFormat(string formatSpecifier)
         {
+            List<ExceptionRenderingFormat> formats = new List<ExceptionRenderingFormat>();
             string[] parts = formatSpecifier.Replace(" ", string.Empty).Split(',');
-            var dataTargets = new List<ExceptionDataTarget>();
 
             foreach (string s in parts)
             {
+                ExceptionRenderingFormat renderingFormat;
+
                 switch (s.ToUpper(CultureInfo.InvariantCulture))
                 {
                     case "MESSAGE":
-                        dataTargets.Add(AppendMessage);
+                        renderingFormat = ExceptionRenderingFormat.Message;
                         break;
-
                     case "TYPE":
-                        dataTargets.Add(AppendType);
+                        renderingFormat = ExceptionRenderingFormat.Type;
                         break;
-
                     case "SHORTTYPE":
-                        dataTargets.Add(AppendShortType);
+                        renderingFormat = ExceptionRenderingFormat.ShortType;
                         break;
-
                     case "TOSTRING":
-                        dataTargets.Add(AppendToString);
+                        renderingFormat = ExceptionRenderingFormat.ToString;
                         break;
-
                     case "METHOD":
-                        dataTargets.Add(AppendMethod);
+                        renderingFormat = ExceptionRenderingFormat.Method;
                         break;
-
                     case "STACKTRACE":
-                        dataTargets.Add(AppendStackTrace);
+                        renderingFormat = ExceptionRenderingFormat.StackTrace;
                         break;
-
                     case "DATA":
-                        dataTargets.Add(AppendData);
+                        renderingFormat = ExceptionRenderingFormat.Data;
                         break;
-
                     default:
                         InternalLogger.Warn("Unknown exception data target: {0}", s);
+                        renderingFormat = ExceptionRenderingFormat.NotSet;
```

Comment:
Proposal: remove `NotSet` and change `ExceptionRenderingFormat renderingFormat` to `ExceptionRenderingFormat? renderingFormat`